src/main/java/org/eclipse/tractusx/managedidentitywallets/service/PresentationService.java

```diff
@@ -121,39 +117,34 @@ public Map<String, Object> createPresentation(Map<String, Object> data, boolean
             verifiableCredentials.add(verifiableCredential);
         });
 
+        SigningService keyStorageService = availableSigningServices.get(callerWallet.getSigningServiceType());
+        PresentationCreationConfig presentationConfig = null;
         Map<String, Object> response = new HashMap<>();
+        Did vpIssuerDid = DidParser.parse(callerWallet.getDid());
         if (asJwt) {
             log.debug("Creating VP as JWT for bpn ->{}", callerBpn);
             Validate.isFalse(StringUtils.hasText(audience)).launch(new BadDataException("Audience needed to create VP as JWT"));
-
             //Issuer of VP is holder of VC
-            Did vpIssuerDid = DidParser.parse(callerWallet.getDid());
-
-            //JWT Factory
-            SerializedJwtPresentationFactory presentationFactory = new SerializedJwtPresentationFactoryImpl(
-                    new SignedJwtFactory(new OctetKeyPairFactory()), new JsonLdSerializerImpl(), vpIssuerDid);
-
-            //Build JWT
-            x21559PrivateKey ed25519Key = walletKeyService.getPrivateKeyByWalletIdentifier(callerWallet.getId());
-            x21559PrivateKey privateKey = new x21559PrivateKey(ed25519Key.asByte());
-            SignedJWT presentation = presentationFactory.createPresentation(vpIssuerDid
-                    , verifiableCredentials, audience, privateKey);
-
-            response.put(StringPool.VP, presentation.serialize());
+            presentationConfig = PresentationCreationConfig.builder()
```

Comment:
a small refactor presentationConfig can be initialized outside the condition with the shared properties and build after the condition.